# Placeholder

In this data analysis project, we take a look at an extensive collection of philisophical text data which span 13 schools of thought. Specifically, we have Plato, Aristotle, Nietzsche, Rationalism, Empiricism, German Idealism, Communism, Capitalism, Phenomenology, Continental Philosophy, Analytic Philosophy,  Feminism, and Stoicism.

In [7]:
import sys
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
print(sys.version)

3.10.4 (main, Mar 31 2022, 03:38:35) [Clang 12.0.0 ]


In [5]:
phi_data = pd.read_csv('philosophy_data.csv') #download csv as a DataFrame

In [6]:
pd.options.display.max_colwidth = 75
phi_data.sample(5)

title     author  \
156854                                       Lewis - Papers      Lewis   
324545  A General Theory Of Employment, Interest, And Money     Keynes   
285056                  Elements Of The Philosophy Of Right      Hegel   
65522                            Aristotle - Complete Works  Aristotle   
316933                                The Wealth Of Nations      Smith   

                 school  \
156854         analytic   
324545       capitalism   
285056  german_idealism   
65522         aristotle   
316933       capitalism   

                                                                    sentence_spacy  \
156854  But there is no second event that is essentially a having of the existe...   
324545  At the cost of anticipating what is more properly the subject of chapte...   
285056  The ethical world, on the other hand, the state, or reason as it actual...   
65522   And every animal has soul, but a plant is imperfect, and, whereas an an...   
316933  It was now reduced and confined to the original purpose of its institut...   

                                                                      sentence_str  \
156854  But there is no second event that is essentially a having of the existe...   
324545  At the cost of anticipating what is more properly the subject of chapte...   
285056  The ethical world, on the other hand, the state, or reason as it actual...   
65522   And every animal has soul, but a plant is imperfect, and, whereas an an...   
316933  It was now reduced and confined to the original purpose of its institut...   

        original_publication_date  corpus_edition_date  sentence_length  \
156854                       1985                 2008              147   
324545                       1936                 2003              197   
285056                       1820                 1991              125   
65522                        -320                 1991              218   
316933                       1776                 2009               75   

                                                                  sentence_lowered  \
156854  but there is no second event that is essentially a having of the existe...   
324545  at the cost of anticipating what is more properly the subject of chapte...   
285056  the ethical world, on the other hand, the state, or reason as it actual...   
65522   and every animal has soul, but a plant is imperfect, and, whereas an an...   
316933  it was now reduced and confined to the original purpose of its institut...   

                                                                     tokenized_txt  \
156854  ['but', 'there', 'is', 'no', 'second', 'event', 'that', 'is', 'essentia...   
324545  ['at', 'the', 'cost', 'of', 'anticipating', 'what', 'is', 'more', 'prop...   
285056  ['the', 'ethical', 'world', 'on', 'the', 'other', 'hand', 'the', 'state...   
65522   ['and', 'every', 'animal', 'has', 'soul', 'but', 'plant', 'is', 'imperf...   
316933  ['it', 'was', 'now', 'reduced', 'and', 'confined', 'to', 'the', 'origin...   

                                                                    lemmatized_str  
156854   but there be no second event that be essentially a having of the exist...  
324545   at the cost of anticipate what be more properly the subject of chapter...  
285056   the ethical world , on the other hand , the state , or reason as -PRON...  
65522    and every animal have soul , but a plant be imperfect , and , whereas ...  
316933   -PRON- be now reduce and confine to the original purpose of -PRON- ins...

# Preliminary Data Exploration

To explain and prepare for potential issues to come for this data analysis, we first explore the data and display information which is relevant to future analysis.

To analyze these philosophical texts, it would help to get a clear picture in how many sentences from each school we have to work with. This way, it will be easier to determine whether we have enough data can come to any conclusions about specific schools relative to other schools.

In [14]:
phi_data.loc[:,"school"].value_counts()

analytic           55425
aristotle          48779
german_idealism    42136
plato              38366
continental        33779
phenomenology      28573
rationalism        22949
empiricism         19931
feminism           18635
capitalism         18194
communism          17958
nietzsche          13548
stoicism            2535
Name: school, dtype: int64

As we can see, each school has at least 10,000 sentences except for stoicism, which has 2,535. For the most part, there is a lot of data to work with for each school. However, we must take care when considering analysis for stoicism because of its relative lack of data.

In [19]:
print(phi_data.loc[1,["sentence_length","sentence_str"]])

sentence_length                                                                       69
sentence_str       Surely you are not prosecuting anyone before the king archon as I am?
Name: 1, dtype: object


Possible questions:
- common topics of interest for schools of thought
- positive/negative expression and viewpoint of the world
- which schools are similar to which/ which conflict the most
- opinions shift over time
- perspective on religion https://realpython.com/python-nltk-sentiment-analysis/

# Different schools' relationship with religion